core/block.ts

```diff
@@ -120,8 +120,12 @@ export class Block implements IASTNodeLocation, IDeletable {
    * An optional serialization method for defining how to serialize the
    * block's extra state (eg mutation state) to something JSON compatible.
    * This must be coupled with defining `loadExtraState`.
+   *
+   * @param doFullSerialization Whether or not to serialize the full state of
+   *     the extra state (rather than possibly saving a reference to some
+   *     state). This is used during copy-paste.
```

Comment:
nit: could you say which way this should be used in copy-paste? e.g. "This should be `true` when pasting a block." Actually it might be helpful to have more documentation on this parameter because I'm not sure how this is meant to be used. If I am creating custom blocks, I need more information on what is "full" serialization and what isn't. Maybe you can add more documentation to devsite and then link to it here.

Reply:
Added links under blocks and fields. Can't add links under icons because the base icon class doesn't implement serialization - subclasses need to implement the `ISerializable` interface.

The links will be dead until the next release + release of the documentation.